### ISSUE TITLE
Style/RedundantLineContinuation false positive

### ISSUE THREAD

**Issue by jdelStrother** [rubocop/rubocop#11783]
I'm seeing a number of false positives reported by RedundantLineContinuation in 1.50.0 - eg 

```ruby
expect {
  worker.perform
}.to change(ClipImport, :count).by(3).and \
  not_change(AudioClip, :count)
```

```ruby
XPath.descendant(:"trix-editor").where \
  XPath.attr(:id).equals(locator)

```

--------

## Expected behavior

Style/RedundantLineContinuation doesn't warn on the above code.

## Actual behavior

Style/RedundantLineContinuation warns on the above code, and autocorrect tries to fix it by removing the `\`.

```
For /Users/jon/Developer/web: configuration from /Users/jon/Developer/web/.rubocop.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-capybara-2.17.1/config/default.yml
Default configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-1.50.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-capybara-2.17.1/lib/../config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-performance-1.17.1/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-performance-1.17.1/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-rails-2.19.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-rails-2.19.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-rspec-2.19.0/config/default.yml
configuration from /Users/jon/Developer/web/vendor/bundle/nix-bundle/ruby/3.1.0/gems/rubocop-rspec-2.19.0/config/default.yml
Inheriting configuration from /Users/jon/Developer/web/.rubocop_todo.yml
Inspecting 1 file
Scanning /Users/jon/Developer/web/foo.rb
Loading cache from /Users/jon/.cache/rubocop_cache/02b1ead9d63b796a28453205933c8584967ef383/6fe223d0911b45d6ba12dfe334f169f487534acf/fe777f0a0bdd6777443f6052088d110afed2026a
C

Offenses:

foo.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
expect {
^
foo.rb:3:43: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
}.to change(ClipImport, :count).by(3).and \ ...
                                          ^
foo.rb:6:40: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
XPath.descendant(:"trix-editor").where \ ...
                                       ^

1 file inspected, 3 offenses detected, 3 offenses autocorrectable
Finished in 0.41214599926024675 seconds
```

## Steps to reproduce the problem

End a line with a method name and a backslash, then arguments to that method on the next line.

## RuboCop version

```
$ bundle exec rubocop -V
1.50.0 (using Parser 3.2.2.0, rubocop-ast 1.28.0, running on ruby 3.1.2) [arm64-darwin22]
  - rubocop-capybara 2.17.1
  - rubocop-performance 1.17.1
  - rubocop-rails 2.19.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.19.0
```

**Comment by dmke** [comment]
Looks to me like #11752.

**Comment by jdelStrother** [comment]
Given foo.rb : 

```ruby
require "xpath"

xpath = XPath.descendant(:"trix-editor").where \
  XPath.attr(:id).equals("foo")

pp xpath
```

rubocop will autocorrect it to:

```ruby
require "xpath"

xpath = XPath.descendant(:"trix-editor").where
XPath.attr(:id).equals("foo")

pp xpath
```



The former is equivalent to:

```ruby
require "xpath"

xpath = XPath.descendant(:"trix-editor").where(XPath.attr(:id).equals("foo"))

pp xpath
```

the latter is:

```ruby
require "xpath"

xpath = XPath.descendant(:"trix-editor").where()
XPath.attr(:id).equals("foo"))

pp xpath
```

which would raise on trying to call where() - `xpath/dsl.rb:45:in `where': wrong number of arguments (given 0, expected 1)`

**Comment by koic** [comment]
I see. This is an unexpected behavior and I opened https://github.com/rubocop/rubocop/pull/11784 to fix it. Thank you.
